src/main/java/de/sakpaas/backend/service/LocationService.java

```diff
@@ -273,4 +283,34 @@ private void createNewLocation(OsmResultLocationListDto.OsmResultLocationDto osm
     );
     this.save(location);
   }
+
+  /**
+   * Searches in the Nominatim Microservice for the given key.
+   *
+   * @param key The search parameter. Multiple words are separated with %20.
+   * @return The list of Locations in our database
+   */
+  public List<Location> search(String key) {
+    // Makes a request to the Nominatim Microservice
+    final String url = this.searchApiUrl + "/search/" + key + "?format=json";
+    LOGGER.info("URL:" + url);
+    RestTemplate restTemplate = new RestTemplate();
+    ResponseEntity<NominatimSearchResultListDto> response =
+        restTemplate.getForEntity(url, NominatimSearchResultListDto.class);
+
+    List<Location> responseList = new ArrayList<>();
+
+    if (response.getBody() == null) {
+      return responseList;
```

Comment:
```suggestion
      return Collections.emptyList();
```
macht es finde ich deutlicher beim lesen